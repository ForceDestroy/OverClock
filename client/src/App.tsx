import { BrowserRouter, Route, Routes, useLocation } from 'react-router-dom';
import { Box, Grommet } from 'grommet';
import './App.css';

import Home from './pages/Home';
import NotFound from './pages/NotFound';
import Login from './pages/Login';
import EmployeeDashboard from './pages/EmployeeDashboard';
import EmployerDashboard from './pages/EmployerDashboard';
import NavigationTopBar from './components/NavigationTopBar';
import ProfilePage from './pages/ProfilePage';
import Timesheets from './pages/Timesheets';
import Schedule from './pages/Schedule';
import TimeOff from './pages/TimeOff';
import EmployerTimesheets from './pages/EmployerTimesheets';
import Payslip from './pages/Payslip';
import EmployerRequests from './pages/EmployerRequests';
import CompanyPolicy from './pages/CompanyPolicy';

export function App() {
  const { pathname } = useLocation();
  const toggleNav = pathname === '/Login' || pathname === '/';
  const availableRoutes = [
    '/employeedashboard',
    '/employeraccount',
    '/employerdashboard',
    '/employer/timesheets',
    '/employer/requests',
    '/application',
    '/applicationreferral',
    '/applicationmanagement',
    '/companydashboard',
    '/employeedashboard/companypolicy',
    '/profilepage',
    '/employeedashboard/timesheets',
    '/employeedashboard/schedule',
    '/employeedashboard/payslip',
    '/employeedashboard/timeoff',
  ];
  const errorRoute = availableRoutes.includes(pathname.toLowerCase());
  return (
    <div className="App">
      <Grommet full>
        {!toggleNav && errorRoute && <NavigationTopBar />}
        <Box direction="row" height={{ min: '90vh' }} background="light-1">
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/Login" element={<Login />} />
            <Route path="/EmployeeDashboard" element={<EmployeeDashboard />} />
            <Route path="/EmployerDashboard" element={<EmployerDashboard />} />
            <Route path="/Employer/Requests" element={<EmployerRequests />} />
            <Route
              path="/Employer/Timesheets"
              element={<EmployerTimesheets />}
            />
            <Route
              path="/EmployeeDashboard/CompanyPolicy"
              element={<CompanyPolicy />}
            />
            <Route path="/ProfilePage" element={<ProfilePage />} />
            <Route
              path="/EmployeeDashboard/Timesheets"
              element={<Timesheets />}
            />
            <Route path="/EmployeeDashboard/Schedule" element={<Schedule />} />
            <Route path="/EmployeeDashboard/Payslip" element={<Payslip />} />
            <Route path="/EmployeeDashboard/TimeOff" element={<TimeOff />} />
            <Route path="*" element={<NotFound />} />
          </Routes>
        </Box>
      </Grommet>
    </div>
  );
}

export function WrappedApp() {
  return (
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
}
