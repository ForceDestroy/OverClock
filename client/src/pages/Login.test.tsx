import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { describe, it } from 'vitest';

import { App } from '../App';

describe('Login', () => {
  it('Renders Login to match snapshot', () => {
    // Arrange
    const snap = render(
      <MemoryRouter initialEntries={['/Login']}>
        <App />
      </MemoryRouter>
    );

    // Assert
    expect(snap).toMatchSnapshot();
  });
  it('Renders LoginBox', () => {
    // Arrange
    render(
      <MemoryRouter initialEntries={['/Login']}>
        <App />
      </MemoryRouter>
    );

    // Act
    // No Act for this Test

    // Assert
    expect(screen.getByAltText('login_image')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Email')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Password')).toBeInTheDocument();
    expect(screen.getByText('Login')).toBeInTheDocument();
  });
});
