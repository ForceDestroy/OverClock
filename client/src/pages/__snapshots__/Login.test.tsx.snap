// Vitest Snapshot v1

exports[`Login > Renders Login to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="StyledGrommet-sc-19lkkz7-0 jGVJhU"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 eETalo"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 jaXBUd"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 eEByvQ"
              >
                <h1
                  class="StyledHeading-sc-1rdh4aw-0 ehosPX"
                >
                  Login
                </h1>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 qdAYv"
                />
                <img
                  alt="login_image"
                  class="StyledImage-sc-ey4zx9-0 iAcpfl"
                  src="/src/assets/images/loginImage.png"
                />
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 qdAYv"
                />
                <div
                  class="StyledBox-sc-13pk1d4-0 bfWDjh"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 emSMsc"
                  >
                    <form>
                      <div
                        class="StyledBox-sc-13pk1d4-0 iSXGxe"
                      >
                        <div
                          class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                        >
                          <div
                            class="StyledTextInput__StyledIcon-sc-1x30a0s-3 hTpbjn"
                          >
                            <svg
                              aria-label="MailOption"
                              class="StyledIcon-sc-ofa7kd-0 kKhdEN"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M1 4h22v16H1V4zm0 1 11 8.5L23 5"
                                fill="none"
                                stroke="#000"
                                stroke-width="2"
                              />
                            </svg>
                          </div>
                          <input
                            autocomplete="off"
                            class="StyledTextInput-sc-1x30a0s-0 bISior"
                            placeholder="Email"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class="StyledBox-sc-13pk1d4-0 gBxcJd"
                      >
                        <div
                          class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                        >
                          <div
                            class="StyledTextInput__StyledIcon-sc-1x30a0s-3 hTpbjn"
                          >
                            <svg
                              aria-label="Key"
                              class="StyledIcon-sc-ofa7kd-0 kKhdEN"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M10 13v3h3v3h3v2l2 2h5v-4L12.74 8.74C12.91 8.19 13 7.6 13 7c0-3.31-2.69-6-6-6S1 3.69 1 7a6.005 6.005 0 0 0 8.47 5.47L10 13zM6 7a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
                                fill="none"
                                stroke="#000"
                                stroke-width="2"
                              />
                            </svg>
                          </div>
                          <input
                            autocomplete="off"
                            class="StyledTextInput-sc-1x30a0s-0 bISior"
                            placeholder="Password"
                            type="password"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class="StyledBox-sc-13pk1d4-0 fvAWnZ"
                      >
                        <button
                          class="StyledButton-sc-323bzc-0 jMQIxB"
                          data-testid="loginButton"
                          type="button"
                        >
                          <div
                            class="StyledBox-sc-13pk1d4-0 dfdKQJ"
                          >
                            <br />
                            <span
                              class="StyledText-sc-1sadyjn-0 cKTRLX"
                            >
                              Log In
                            </span>
                            <br />
                          </div>
                        </button>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="StyledGrommet-sc-19lkkz7-0 jGVJhU"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 eETalo"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 jaXBUd"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 eEByvQ"
            >
              <h1
                class="StyledHeading-sc-1rdh4aw-0 ehosPX"
              >
                Login
              </h1>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 qdAYv"
              />
              <img
                alt="login_image"
                class="StyledImage-sc-ey4zx9-0 iAcpfl"
                src="/src/assets/images/loginImage.png"
              />
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 qdAYv"
              />
              <div
                class="StyledBox-sc-13pk1d4-0 bfWDjh"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 emSMsc"
                >
                  <form>
                    <div
                      class="StyledBox-sc-13pk1d4-0 iSXGxe"
                    >
                      <div
                        class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                      >
                        <div
                          class="StyledTextInput__StyledIcon-sc-1x30a0s-3 hTpbjn"
                        >
                          <svg
                            aria-label="MailOption"
                            class="StyledIcon-sc-ofa7kd-0 kKhdEN"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M1 4h22v16H1V4zm0 1 11 8.5L23 5"
                              fill="none"
                              stroke="#000"
                              stroke-width="2"
                            />
                          </svg>
                        </div>
                        <input
                          autocomplete="off"
                          class="StyledTextInput-sc-1x30a0s-0 bISior"
                          placeholder="Email"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="StyledBox-sc-13pk1d4-0 gBxcJd"
                    >
                      <div
                        class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                      >
                        <div
                          class="StyledTextInput__StyledIcon-sc-1x30a0s-3 hTpbjn"
                        >
                          <svg
                            aria-label="Key"
                            class="StyledIcon-sc-ofa7kd-0 kKhdEN"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M10 13v3h3v3h3v2l2 2h5v-4L12.74 8.74C12.91 8.19 13 7.6 13 7c0-3.31-2.69-6-6-6S1 3.69 1 7a6.005 6.005 0 0 0 8.47 5.47L10 13zM6 7a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
                              fill="none"
                              stroke="#000"
                              stroke-width="2"
                            />
                          </svg>
                        </div>
                        <input
                          autocomplete="off"
                          class="StyledTextInput-sc-1x30a0s-0 bISior"
                          placeholder="Password"
                          type="password"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="StyledBox-sc-13pk1d4-0 fvAWnZ"
                    >
                      <button
                        class="StyledButton-sc-323bzc-0 jMQIxB"
                        data-testid="loginButton"
                        type="button"
                      >
                        <div
                          class="StyledBox-sc-13pk1d4-0 dfdKQJ"
                        >
                          <br />
                          <span
                            class="StyledText-sc-1sadyjn-0 cKTRLX"
                          >
                            Log In
                          </span>
                          <br />
                        </div>
                      </button>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
